packages/kokoas-e2e/cypress/e2e/contractPreview2/input.cy.ts

```diff
@@ -0,0 +1,204 @@
+          .invoke('text')
+          .then((textCost) => {
+            const cost = Number(textCost.replace(/[^0-9]/g, ''));
+            cy.wrap(cost)
+              .should('eq', Math.round(taxExcludedAmt - totalProfit));
+            return cy.wrap({
+              taxExcludedAmt,
+              cost,
+            });
+          }))
+        .then(({
+          taxExcludedAmt,
+          cost,
+        }) => cy
+          .contains('span', '粗利率')
+          .siblings('span')
+          .invoke('text')
+          .then((textProfitRate) => {
+          // get profitRate with decimal point
+            const profitRate = Number(textProfitRate.replace(/[^0-9.]/g, ''));
+            console.log(taxExcludedAmt, cost, profitRate);
+         
+            /** 利益率 = ( 単価 - 原価) / 単価 */
+            const calculatedTaxRate = ((taxExcludedAmt - cost) /  taxExcludedAmt) * 100;
+        
+            cy.wrap(profitRate.toFixed(2))
+              .should('eq', calculatedTaxRate.toFixed(2));
+          }));
+    });
+
+    context(
+      '支払いの挙動', 
+      {
+        testIsolation: false,
+      }, 
+      () => {
+        const dividedAmt = Math.round(totalContractAmt / 4);
+        const payments = [
+          ['契約金', 'contract'],
+          ['着手金', 'initial'],
+          ['中間金', 'interim'],
+          ['最終金', 'final'],
+        ];
+
+        it('契約金すること', () => {
+          cy.getTextInputsByLabel('契約合計金額')
+            .type(totalContractAmt.toString())
+            .blur()
+            .should('value', totalContractAmt.toLocaleString());
+        });
+
+        payments
+          .map(([label, namePart], idx) => {
+            it(`${label} - チェックが入ると残り金額が生成される`, () => {
+              cy.log('初期状態はチェックが入っていないこと');
+              cy.get(`input[name="${namePart}Amt"]`)
+                .as('amt')
+                .should('value', '0');
+              cy.get(`input[name="${namePart}AmtDate"]`)
+                .as('dt')
+                .should('value', '');
+
+              const expectedRemainingAmt = totalContractAmt - (dividedAmt * idx);
+
+              cy.log(`${label} - チェックボックスの挙動`);
+              cy.getCheckboxesByLabel(label).check();
+              cy.get('@amt').should('value', expectedRemainingAmt.toLocaleString());
+              cy.get('@dt').should('value', '');
+
+              cy.get('@amt').clear()
+                .type(dividedAmt.toString(), { scrollBehavior: 'center' })
+                .blur();
+
+              cy.get('@amt').should('value', dividedAmt.toLocaleString());
+              cy.get('@dt').type(format( addMonths(new Date(), idx), 'yyyy/MM/dd'));
+            
+            });
+          });
+
+        payments.map(([label, namePart]) => {
+          it(`${label} - チェックを外すと残り金額が0になる`, () => {
+            cy.log(`${label} - チェックを外すと残り金額が0になる`);
+            cy.getCheckboxesByLabel(label).uncheck();
+            cy.get(`input[name="${namePart}Amt"]`)
+              .should('value', '0');
+          });
+        });
```

Comment:
```suggestion
        payments.map(([label, namePart]) => {
          it(`${label} - チェックを外すと対象の金額が0になる`, () => {
            cy.log(`${label} - チェックを外すと対象の金額が0になる`);
            cy.getCheckboxesByLabel(label).uncheck();
            cy.get(`input[name="${namePart}Amt"]`)
              .should('value', '0');
          });
        });
```

「残り金額」だと、残金(remaining balance？)のことになってしまい、実態と一致しないかなと思いました。
修正案を出しましたので、確認お願いします